examples/reducestream/sum/example.py

```diff
@@ -0,0 +1,49 @@
+import logging
+import os
+from collections.abc import AsyncIterable
+
+from pynumaflow import setup_logging
+from pynumaflow.reducestreamer import (
+    Message,
+    Datum,
+    Metadata,
+    ReduceStreamAsyncServer,
+    ReduceStreamer,
+)
+from pynumaflow.shared.asynciter import NonBlockingIterator
+
+_LOGGER = setup_logging(__name__)
+if os.getenv("PYTHONDEBUG"):
+    _LOGGER.setLevel(logging.DEBUG)
+
+
+class ReduceSum(ReduceStreamer):
+    def __init__(self, counter):
+        self.counter = counter
+
+    async def handler(
+        self,
+        keys: list[str],
+        datums: AsyncIterable[Datum],
+        output: NonBlockingIterator,
+        md: Metadata,
+    ):
+        async for msg in datums:
+            val = int(msg.value)
+            self.counter += val
+            if self.counter >= 100:
+                msg = f"{self.counter}"
+                await output.put(Message(str.encode(msg), keys=keys))
```

Comment:
add a comment on how the user experience will be if you stream results early on.